### ISSUE TITLE
Add TypeScript definition

### ISSUE THREAD

**Issue by PavelPZ** [MrSwitch/hello.js#46]
Thanks a lot for this amazing library.

Have you any plan to provide a Typescript definition for hello.js? It simplified ids integration to large projects.

Thanks advanced

**Comment by pra2006** [comment]
Hey,
Loving HelloJS it appears that the typescript d.ts hasn't been updated for some time and isn't aware of ng.IPromise

**Comment by Sn3b** [comment]
Hey guys, same as @pra2006, loving HelloJS, but I fail to find an example to use it in typescript :(
I'd like to add it to an Angular2 app, is there any example out there? All searches on Hello and Angular2 return hello world examples...

**Comment by amigup** [comment]
Hello @PavelPZ ,
Can you please provide the updated typescript? Also, I'd like to use the library in anulgar4 application, do you have any usage example of hellojs typescript?

**Comment by vuorinem** [comment]
I went through the code and the documentation, and updated the type definitions as well as I could. Any feedback is welcome :)

https://github.com/DefinitelyTyped/DefinitelyTyped/pull/18318

**Comment by baywet** [comment]
Ideally those type definitions should be moved to a /types folder and the package.json should be updated to contain a types entry. DefinitelyTyped is only here for repos that "don't want the types".

**Comment by hhubik** [comment]
@vuorinem  To authenticate an Angular SPA against MS Azure B2C, I need to specify the `response_type` to be `token id_token`. This not allowed by the type `HelloJSTokenResponseType` definition in version 1.16.0.

**Comment by vuorinem** [comment]
@hhubik good point, I probably mistook the "example values" from the documentation as the list of all possible values.

Do you think the list of [OAuth Authorization Endpoint Response Types](https://www.iana.org/assignments/oauth-parameters/oauth-parameters.xhtml#endpoint) would work, or should the type or response_type rather be just a string? One problem with the list is that the type would probably need to specify all the possible orders of the parameters (e.g. "id_token token" and "token id_token").

**Comment by hhubik** [comment]
@vuorinem My preference would be to replicate the entries from the "OAuth Authorization Endpoint Response Types" list.  I consider "id_token token" and "token id_token" to be "functionally" identical.  Therefore I would include only one of them, probably the first one where the types are in alphabetical order, which is what the IANA folks appear to have done.

**Comment by vuorinem** [comment]
Created a pull request: https://github.com/DefinitelyTyped/DefinitelyTyped/pull/24672
